### ISSUE TITLE
Deferred doesn't participate in QoS Overrides

### ISSUE THREAD

**Issue by zwaldowski** [bignerdranch/Deferred#10]
From WWDC 2015 Session 212:

![screen shot 2015-07-13 at 1 43 42 pm](https://cloud.githubusercontent.com/assets/170812/8656377/347a057c-2965-11e5-8540-7aaf54eea1b6.png)

Deferred currently uses `dispatch_group_wait` and `dispatch_async` for chaining, which can exclude participation in QoS propagation and contribute to priority inversion issues. Use of the `dispatch_block` APIs should be investigated for `Deferred` chaining.

**Comment by zwaldowski** [comment]
Since `value` uses `dispatch_block_wait` in certain implementations, this is resolved for that in #13 and #30 for Swift 1 and 2, respectively.

Supporting it for `upon` would be a trivial amount of extra code. However, for Swift 2.0, that is currently blocked by rdar://22437691 ([OpenRadar link](http://www.openradar.me/radar?id=5055516279570432)), and can't be easily fixed in the same way as in #30.

**Comment by zwaldowski** [comment]
Resolved for Swift 2.0 (incl. workarounds) in #30 and #33.

**Comment by lyricsboy** [comment]
Can we close this now that #30 is merged?

**Comment by zwaldowski** [comment]
Indeed!
